lite/inquiring_certifier.go

```diff
@@ -1,163 +1,216 @@
 package lite
 
 import (
-	"github.com/tendermint/tendermint/types"
+	"bytes"
 
-	liteErr "github.com/tendermint/tendermint/lite/errors"
+	lerr "github.com/tendermint/tendermint/lite/errors"
+	"github.com/tendermint/tendermint/types"
+	log "github.com/tendermint/tendermint/libs/log"
 )
 
 var _ Certifier = (*InquiringCertifier)(nil)
 
-// InquiringCertifier wraps a dynamic certifier and implements an auto-update strategy. If a call
-// to Certify fails due to a change it validator set, InquiringCertifier will try and find a
-// previous FullCommit which it can use to safely update the validator set. It uses a source
-// provider to obtain the needed FullCommits. It stores properly validated data on the local system.
+// InquiringCertifier implements an auto-updating certifier.  It uses a
+// "source" provider to obtain the needed FullCommits to securely sync with
+// validator set changes.  It stores properly validated data on the
+// "trusted" local system.
 type InquiringCertifier struct {
-	cert *DynamicCertifier
-	// These are only properly validated data, from local system
-	trusted Provider
-	// This is a source of new info, like a node rpc, or other import method
-	Source Provider
+	logger  log.Logger
+	chainID string
+	// These are only properly validated data, from local system.
+	trusted PersistentProvider
+	// This is a source of new info, like a node rpc, or other import method.
+	source Provider
 }
 
-// NewInquiringCertifier returns a new Inquiring object. It uses the trusted provider to store
-// validated data and the source provider to obtain missing FullCommits.
+// NewInquiringCertifier returns a new InquiringCertifier. It uses the
+// trusted provider to store validated data and the source provider to
+// obtain missing data (e.g. FullCommits).
 //
-// Example: The trusted provider should a CacheProvider, MemProvider or files.Provider. The source
-// provider should be a client.HTTPProvider.
-func NewInquiringCertifier(chainID string, fc FullCommit, trusted Provider,
-	source Provider) (*InquiringCertifier, error) {
-
-	// store the data in trusted
-	err := trusted.StoreCommit(fc)
-	if err != nil {
-		return nil, err
-	}
-
+// The trusted provider should a CacheProvider, MemProvider or
+// files.Provider.  The source provider should be a client.HTTPProvider.
+func NewInquiringCertifier(chainID string, trusted PersistentProvider, source Provider) *InquiringCertifier {
```

Comment:
Why it does not return Certifier?

Reply:
It's better to return concrete types, whereas arguments are often better as interfaces.